modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -479,7 +480,7 @@ const DomParser = (settings: DomParserSettings = {}, schema = Schema()): DomPars
    */
   const parse = (html: string, args: ParserArgs = {}): AstNode => {
     const validate = defaultedSettings.validate;
-    const preferFullDocument = defaultedSettings.root_name === '#document';
+    const preferFullDocument = (args.root_name ?? defaultedSettings.root_name) === '#document';
```

Comment:
We don't need a new option you should be able to check if `args.context` is set to `#document` then you do a full page parse. That was what the old parser did.

So I guess this:
```suggestion
    const preferFullDocument = (args.context ?? defaultedSettings.root_name) === '#document';
```

Reply:
I was concerned about this creating unintended side-effects since `context` is used in different places than `root_name`, it seems to change the validation steps.

Reply:
I suppose that if it turns out to be a problem it won't be that hard to split it out.